### PR TITLE
Small cleanup of MockRegistryClient

### PR DESCRIPTION
**Pull request by survivorbat**
- Add a Codec so that you can actually replicate the example
- Remove unused codecCreationEnabled
- Change `schema := &Schema{}` to `var schema *Schema` because the former can never be nil in the == nil check
- Update tests to increase coverage and prevent issues like above
- Change version fetch

Related to #92

### CODE REVIEW

File: mockSchemaRegistryClient_test.go [hunk 1]
```diff
@@ -1,146 +1,595 @@
...
-func TestMockSchemaRegistryClient_DeleteSubjectByVersion(t *testing.T) {
-	originalCount, _ := srClient.GetSchemaVersions("test1-value")
-	err := srClient.DeleteSubjectByVersion("test1-value", 1, true)
-	assert.NoError(t, err)
-	newCount, _ := srClient.GetSchemaVersions("test1-value")
-	assert.Equal(t, len(originalCount)-1, len(newCount))
+var avroType = Avro
+var protobuf = Protobuf
+
+func TestMockSchemaRegistryClient_CreateSchema_RegistersSchemaCorrectly(t *testing.T) {
```
Comment (AtakanColak) [inline mockSchemaRegistryClient_test.go:11]: Is it possible to reduce code repetition in schema definitions?

**Review by AtakanColak — CHANGES_REQUESTED** [review]

File: mockSchemaRegistryClient_test.go (hunk 1 above)
Comment (survivorbat) [reply mockSchemaRegistryClient_test.go:18]: Good one! Update :+1:

**Comment by survivorbat** [comment]
Hey @AtakanColak I updated the pull request with the requested changes :)

**Review by AtakanColak — APPROVED** [review]

**Comment by AtakanColak** [comment]
Thanks for your contribution 🎉

**Pull request merged**